src/paradigma/heart_rate/feature_extraction.py

```diff
@@ -0,0 +1,288 @@
+from typing import List, Tuple, Union
+from scipy.signal import find_peaks
+import pandas as pd
+import numpy as np
+from scipy.signal import welch, find_peaks
+from scipy.signal.windows import hamming
+from scipy.stats import kurtosis, skew
+from paradigma.heart_rate.heart_rate_analysis_config import PPGconfig
+
+def generate_statistics(
+        sensor_col: pd.Series,
+        statistic: str
+    ) -> list:
+    """Generate statistics for a single sensor and axis. The function is used with the apply function in pandas.
+    
+    Parameters
+    ----------
+    sensor_col: pd.Series
+        The sensor column to be aggregated (e.g. x-axis of accelerometer)
+    statistic: str
+        The statistic to be computed ['mean', 'var', 'median', 'kurtosis', 'skewness']
+        
+    Returns
+    -------
+    list
+        the statistic for the sensor segments
+    """
+    if statistic == 'mean':
+        return [np.mean(np.abs(x)) for x in sensor_col]
+    elif statistic == 'var':
+        return [np.var(x, ddof=1) for x in sensor_col]  # ddof=1 for unbiased variance is used, same as matlab
+    elif statistic == 'median':
+        return [np.median(np.abs(x)) for x in sensor_col]
+    elif statistic == 'kurtosis':
+        return [kurtosis(x, fisher = False) for x in sensor_col] # Fisher's definition of kurtosis is used, same as matlab
+    elif statistic == 'skewness':
+        return [skew(x) for x in sensor_col]
+    else:
+        raise ValueError("Statistic not recognized.")
+
+def compute_signal_to_noise_ratio(
+        ppg_segments: np.ndarray
+    ) -> list:
+    """
+    Compute the signal to noise ratio of the PPG signal.
+    
+    Args:
+    arr_ppg (numpy.ndarray): PPG signal.
+    
+    Returns:
+    list: Signal to noise ratio of the PPG windows.
+    """
+    l_signal_to_noise_ratios = []
+    for segment in ppg_segments:
+        arr_signal = np.var(segment)
+        arr_noise = np.var(np.abs(segment))
+        signal_to_noise_ratio = arr_signal / arr_noise
+        l_signal_to_noise_ratios.append(signal_to_noise_ratio)
+    
+    return l_signal_to_noise_ratios
```

Comment:
Je kan ook gebruik maken van Numpy arrays hier, dan hoef je niet te loopen.

```suggestion
    arr_signal = np.var(ppg_segments, axis=1)
    abs_signal = np.abs(ppg_segments, axis=1)
    arr_noise = np.var(abs_signal, axis=1)
    signal_to_noise_ratio = arr_signal / arr_noise
        
    return signal_to_noise_ratio
```